compiler/damlc/tests/daml-test-files/SemanticsValue.daml

```diff
@@ -0,0 +1,45 @@
+
+-- | This module checks that what is considered a value in
+-- the operational semantics of DAML-LF matches what is
+-- treated as a value in DAML.
+--
+-- According to the spec, a value is "an expression which does not
+-- need to be evaluated further". Any unevaluated subexpressions
+-- in a value should remain unevaluated. In particular, this
+-- includes:
+--
+-- * term lambdas
+-- * type lambdas where the parameter has kind 'nat'
+-- * updates with 'bind' or 'embed_expr'
+-- * scenarios with 'sbind' or 'sembed_expr'
+--
+-- Where we can, we test this from DAML by using values with
+-- embedded error expressions. The value should evaluate without
+-- raising an error.
+--
+-- Any errors here are tagged with the name of the rule in
+-- the semantics that they violate.
+
+module SemanticsValue where
+
+main = scenario do
```

Comment:
It's hard to test in DAML because of Damas–Milner. :-/

It's also a bit out of scope for this module, which tests that things *are* values, not that they *aren't* values. (Ok, I know that's a weak argument...)

I think we should definitely test this in the DAML-LF version of the spec tests.

Reply:
I had an example in https://github.com/digital-asset/daml/pull/5547#issuecomment-616565362 that exhibits the issue.

Reply:
I added this as test in the other file. Thanks!

Reply:
Thanks for adding it!